types/coin.go

```diff
@@ -508,15 +513,49 @@ func ParseCoin(coinStr string) (coin Coin, err error) {
 
 	denomStr, amountStr := matches[2], matches[1]
 
-	amount, ok := NewIntFromString(amountStr)
-	if !ok {
-		return Coin{}, fmt.Errorf("failed to parse coin amount: %s", amountStr)
-	}
-
 	if denomStr != strings.ToLower(denomStr) {
 		return Coin{}, fmt.Errorf("denom cannot contain upper case characters: %s", denomStr)
 	}
 
+	// test if the amount is specified using scientific notation
+	scinMatches := reScinAmt.FindStringSubmatch(amountStr)
```

Comment:
It would be good to separate out some of the functionality you've built here into some new functions to improve clarity Id imagine two additional functions:
`IsScientificNotation(string) bool` and
`ScientificNotationToInt(string) sdk.Int`
This would make this parent function more lean/clear

Reply:
I have moved most code into `ParseScientificNotation() Int bool error` where `ok bool` returns false if not detected.  I can add `IsScientificNotation(string)` but would result in two calls to the same regexp (as I need the regexp `matches` to parse ...so for now, I opted to go with a single function.